build.sh

```diff
@@ -443,12 +444,12 @@ git_clone() {
 }
 
 update_script_name() {
-    $ECHO `dirname $0`/_update.sh
+    $ECHO "$(dirname $0)/_update.sh"
 }
 
 update_script() {
-    update_script=`update_script_name`
-    bash_path=`which bash`
+    local -r update_script=$(update_script_name)
```

Comment:
Why `local -r` here, but not for pretty much all other variables? I don't see either `update_script` or `bash_path`  used in multiple places, so I don't think this was causing namespace pollution before. Maybe I'm missing something though?

Reply:
`local` should be used as much as possible for variables that don't need to be global -- someone comes along and adds a new function accidentally using the same global names, and it doesn't work.

but aside from here and `clang_version_ok`, there are no other local variables (or variables that could be made local) in this entire file, everything is globals. it works, and i don't want to change that much.

as for `-r`, the same is true, there are no other possible places to add it.  in `clang_version_ok`  the variables have to be mutable (or the function re-designed). 

these are readonly in this function because there's no reason they should need to change after being set.

Reply:
the `-r` is because these shouldn't be changed later in the function, and I added `local` because maybe someone will come along and use those names local in another function, and be surprised their code breaks

Reply:
(it also satisfies the linter Shellcheck who doesn't like `local x=y` for "masking return values", although I did ignore Shellcheck in most places)